Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1166,3 +1166,36 @@ void TextureCache::TCacheEntryBase::Zero(u8* ptr)
 		ptr += memory_stride;
 	}
 }
+
+u64 TextureCache::TCacheEntryBase::CalculateHash() const
+{
+	u8* ptr = Memory::GetPointer(addr);
+	if (memory_stride == CacheLinesPerRow() * 32)
+	{
+		return GetHash64(ptr, size_in_bytes, g_ActiveConfig.iSafeTextureCache_ColorSamples);
+	}
+	else
+	{
+		u32 blocks = NumBlocksY();
+		u64 temp_hash = size_in_bytes;
+
+		u32 samples_per_row;
+		if (g_ActiveConfig.iSafeTextureCache_ColorSamples == 0)
```

Comment:
you can simplify this down to:

    u32 samples_per_row = 0;
    if (g_ActiveConfig.iSafeTextureCache_ColorSamples != 0)
    {
        samples_per_row == std::max(....
    }